src/lib/TranscriptEditor/TimedTextEditor/index.js

```diff
@@ -145,29 +145,48 @@ class TimedTextEditor extends React.Component {
   }
 
   localSave = () => {
-    const mediaUrl = this.props.mediaUrl;
+    let mediaUrlName = this.props.mediaUrl;
+    // if using local media instead of using random blob name
+    // that makes it impossible to retrieve from on page refresh
+    // use file name
+    if (this.props.mediaUrl.includes('blob')) {
```

Comment:
Seems quite arbitrary, is this going to stay as the convention for now? We could have a more solid constant defined at the top of the file so you're not checking 'blob' in so many places?

Reply:
I am going to leave it for now, coz it would need re-thinking how it is communicating with the parent component, eg removing the use of `ref`, but suggestions for that refactor welcome.